iib/workers/tasks/opm_operations.py

```diff
@@ -487,6 +489,133 @@ def _serve_cmd_at_port(
     raise IIBError(f'Index registry has not been initialized after {max_tries} tries')
 
 
+def get_list_packages(
+    input_data: str,
+    base_dir: str,
+) -> List[str]:
+    """
+    Get list of olm.package names from input data.
+
+    :param str input_data: input data for opm render
+        Example: catalog-image | catalog-directory | bundle-image | bundle-directory | sqlite-file
+    :param str base_dir: temp directory where opm will be executed.
+    :return: list of package names present in input data.
+    :rtype: [str]
+    """
+    olm_packages = opm_render(input_data, base_dir)
+
+    package_names = [
+        olm_package['name']
+        for olm_package in olm_packages
+        if olm_package['schema'] == 'olm.package'
+    ]
+
+    return package_names
+
+
+def _get_olm_package_version(olm_bundle: dict) -> str:
+    """
+    Find and return version of OLM bundle.
+
+    :param BundleImage olm_bundle: olm bundle dictionary
+    :return: OLM bundle version
+    :rtype: str
+    """
+    for property in olm_bundle['properties']:
+        if property['type'] == "olm.package":
+            return property['value']['version']
+
+    log.warning("No olm package version found for OLM bundle.")
+    return ""
```

Comment:
Should this raise an error? I don't see the empty value being handled in the rest of the code.

Reply:
Yeah, It might be better. I've updated the code.